thorlcr/thorutil/thbuf.cpp

```diff
@@ -2727,28 +2715,11 @@ class CSharedFullSpillingWriteAhead : public CInterfaceOf<ISharedRowStreamReader
     }
     virtual unsigned __int64 getStatistic(StatisticKind kind) const override
     {
-        StatisticKind useKind;
-        switch (kind)
-        {
-            case StSizeSpillFile:
-                useKind = StSizeDiskWrite;
-                break;
-            case StCycleDiskWriteIOCycles:
-            case StTimeDiskWriteIO:
-            case StSizeDiskWrite:
-                return 0;
-            case StNumSpills:
-                return 1;
-            case StTimeSpillElapsed:
-                useKind = StCycleDiskWriteIOCycles;
-                break;
-            default:
-                useKind = kind;
-        }
-        unsigned __int64 v = 0;
+        if (kind==StNumSpills)
+            return 1;
+        unsigned __int64 v = inactiveStats.getStatisticValue(kind);
         if (likely(iFileIO))
-            v = iFileIO->getStatistic(useKind);
-        v += inactiveStats.getStatisticValue(useKind);
+            v = iFileIO->getStatistic(kind);
```

Comment:
this is potential thread unsafe. iFileIO could be null at this point, i.e. closeWriter could get in between the test and this line
Also shouldn't this still be v += ? inactive + current IFileIO stats?

Reply:
re. the thread safety an alternative is not to examine the potentially changing thread-unsafe iFileIO here at all, 
instead merge the stats into a stats container (e.g. inactive) when they are written, i.e. at the end of writeRowsFromInput().

Reply:
@shamser - this potentially thread unsafe implementation seems to have been reintroduced in latest version